src/components/RichText.ts

```diff
@@ -0,0 +1,161 @@
+        }
+    }
+
+    private static getReadOnlyClasses(readOnlyStyle: ReadOnlyStyle): string {
+        return classNames({
+            "disabled-text": readOnlyStyle === "text",
+            "disabled-bordered": readOnlyStyle === "bordered",
+            "disabled-bordered-toolbar": readOnlyStyle === "borderedToolbar"
+        });
+    }
+
+    private getReadOnlyText(): { __html: string } | undefined {
+        return this.props.readOnly && this.props.readOnlyStyle === "text"
+            ? { __html: sanitizeHtml(this.props.value) }
+            : undefined;
+    }
+
+    private renderQuillNode(): ReactNode {
+        return !(this.props.readOnly && this.props.readOnlyStyle === "text")
+            ? createElement("div", {
+                className: classNames("widget-rich-text-quill"),
+                ref: this.setQuillNode
+            })
+            : null;
+    }
+
+    private setQuillNode(node: HTMLElement) {
+        this.quillNode = node;
+    }
+
+    private setUpEditor(props: RichTextProps) {
+        if (this.quillNode && !this.quill) {
+            this.quill = new Quill(this.quillNode, {
+                modules: this.getEditorOptions(),
+                theme: props.theme
+            });
+
+            this.quill.on("selection-change", this.handleSelectionChange);
+
+            this.updateEditor(props);
+        }
+    }
+
+    private handleSelectionChange() {
+        if (this.quill && !this.quill.hasFocus() && this.props.onChange) {
+            const value = (this.quill as any).container.firstChild.innerHTML;
+            if (this.props.value !== value) {
+                this.props.onChange(value);
+            }
+        }
+    }
+
+    private updateEditor(props: RichTextProps) {
+        if (this.quill) {
+            this.quill.enable(!props.readOnly);
+            this.quill.clipboard.dangerouslyPasteHTML(props.value);
+
+            this.setEditorStyle(props);
+        }
+    }
+
+    private setEditorStyle(props: RichTextProps) {
+        if (this.quillNode) {
+            const quillEditor = this.quillNode.getElementsByClassName("ql-editor")[ 0 ] as HTMLDivElement;
+            if (quillEditor) {
+                quillEditor.classList.add("form-control");
+
+                if (props.minNumberOfLines > 0) {
+                    quillEditor.style.minHeight = `${(props.minNumberOfLines + 1) * this.averageLineHeight}em`;
+                }
+                if (props.maxNumberOfLines > 0) {
+                    quillEditor.style.maxHeight = `${(props.maxNumberOfLines + 1) * this.averageLineHeight}em`;
+                }
+            }
+        }
+    }
+
+    private getEditorOptions(): Quill.StringMap {
+        if (this.props.editorOption === "basic") {
+            return getBasicOptions();
+        }
+        if (this.props.editorOption === "extended") {
+            return getAdvancedOptions();
+        }
+
+        return getToolbar(this.props.customOptions || null);
```

Comment:
`this.props.customOptions` will receive an empty array from Mendix when not items are added in the modeler, so `|| null` will never be called